shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -888,124 +1062,201 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+    // if it's password-related, because it is not in an autofill group.
+    focused = [self getOrCreateAutofillableView:focusedField isPasswordAutofill:isPassword];
+    [_autofillContext removeObjectForKey:focusedId];
+  }
 
-      NSString* autofillId = uniqueIdFromDictionary(field);
-      newInputView.autofillId = autofillId;
+  for (NSDictionary* field in fields) {
+    NSString* autofillId = autofillIdFromDictionary(field);
+    NSAssert(autofillId, @"autofillId must not be null for field: %@", field);
 
-      if ([clientUniqueId isEqualToString:autofillId]) {
-        _activeView = newInputView;
-      }
+    BOOL hasHints = autofillTypeOf(field) != FlutterAutofillTypeNone;
+    BOOL isFocused = [focusedId isEqualToString:autofillId];
 
-      [FlutterTextInputPlugin setupInputView:newInputView withConfiguration:field];
-      [keyWindow addSubview:newInputView];
+    if (isFocused) {
+      focused = [self getOrCreateAutofillableView:field isPasswordAutofill:isPassword];
+    }
+
+    if (hasHints) {
+      // Push the current input field to the context if it has hints.
+      _autofillContext[autofillId] = isFocused ? focused
+                                               : [self getOrCreateAutofillableView:field
+                                                                isPasswordAutofill:isPassword];
+    } else {
+      // Mark for deletion;
+      [_autofillContext removeObjectForKey:autofillId];
     }
   }
 
-  [_activeView setTextInputClient:client];
-  [_activeView reloadInputViews];
-}
+  NSAssert(focused, @"The current focused input view must not be nil.");
+  return focused;
+}
+
+// Returns a new non-reusable input view (and put it into the view hierarchy), or get the
+// view from the current autofill context, if an input view with the same autofill id
+// already exists in the context.
+// This is generally used for input fields that are autofillable (UIKit tracks these veiws
+// for autofill purposes so they should not be reused for a different type of views).
+- (FlutterTextInputView*)getOrCreateAutofillableView:(NSDictionary*)field
+                                  isPasswordAutofill:(BOOL)needsPasswordAutofill {
+  NSString* autofillId = autofillIdFromDictionary(field);
+  FlutterTextInputView* inputView = _autofillContext[autofillId];
+  if (!inputView) {
+    inputView =
+        needsPasswordAutofill ? [FlutterSecureTextInputView alloc] : [FlutterTextInputView alloc];
+    inputView = [[inputView init] autorelease];
+    [self addToKeyWindowIfNeeded:inputView];
+  }
 
-+ (void)setupInputView:(FlutterTextInputView*)inputView
-     withConfiguration:(NSDictionary*)configuration {
-  NSDictionary* inputType = configuration[@"inputType"];
-  NSString* keyboardAppearance = configuration[@"keyboardAppearance"];
-  NSDictionary* autofill = configuration[@"autofill"];
+  inputView.textInputDelegate = _textInputDelegate;
+  [inputView configureWithDictionary:field];
+  return inputView;
```

Comment:
super nit: move the autorelease to the return statement to make auditing a bit easier

Reply:
the `inputView` is not necessarily a new object,  putting `autorelease` at the end would make it a zombie when it's reused.